smf/nexus/multi-sled/config-partial.toml

```diff
@@ -65,6 +65,10 @@ abandoned_vmm_reaper.period_secs = 60
 saga_recovery.period_secs = 600
 lookup_region_port.period_secs = 60
 instance_updater.period_secs = 30
+# How frequently to attempt to restart Failed instances?
+instance_reincarnation.period_secs = 60
+# Default cooldown period between automatic restarts for Failed instances.
+instance_reincarnation.default_cooldown_secs = 60
```

Comment:
Yeah, I think making this cooldown period longer seems totally reasonable --- in fact, I was hoping to get your advice on that! I had just picked one minute kinda arbitrarily. 

Your concern about an instance misbehaving in a way that takes down the whole sled definitely makes sense and I think it's a compelling reason to make the cooldown period much longer. As a contrasting perspective, though, here's the kind of scenario that _I_ was worrying about when I thought that we probably shouldn't make this period too _long_: suppose an instance happens to hit some bug that makes Propolis panic. Maybe in this case that issue isn't too likely to occur again and was some kind of freak accident. We restart that instance, and it ends up on a sled that that bounces for some unrelated reason --- not the fault of the instance. It seems like it wouldn't be great to say "sorry, some service your system relies on is now in Naughty VM Jail for an hour because we decided to treat it like a repeat offender just because it happened to be in the wrong place at the wrong time"...

Maybe the ideal thing is for the cooldown period to be enforced separately for the different reasons an instance might have failed, so that we punish it much more harshly if it does the same thing again than if it happens to get unlucky. But, naively, that means tracking a potentially-variable number of timestamps in CRDB...

Reply:
> "sorry, some service your system relies on is now in Naughty VM Jail for an hour because we decided to treat it like a repeat offender just because it happened to be in the wrong place at the wrong time".

To be fair, it was in the wrong place at the wrong time twice in this situation :)  But yes, we may unfairly penalize some instances sometimes and we could make more sophisticated mechanisms in the future.